pkg/blockchain/ratesAdmin_test.go

```diff
@@ -46,21 +50,15 @@ func TestAddRates(t *testing.T) {
 
 	var err error
 
-	require.Eventually(t, func() bool {
-		err = ratesAdmin.AddRates(context.Background(), rates)
-		return err == nil
-	}, 500*time.Millisecond, 50*time.Millisecond)
+	err = ratesAdmin.AddRates(context.Background(), rates)
 	require.NoError(t, err)
 
 	var returnedRates struct {
 		Rates   []ratesmanager.RatesManagerRates
 		HasMore bool
 	}
 
-	require.Eventually(t, func() bool {
-		returnedRates, err = ratesAdmin.contract.GetRates(&bind.CallOpts{}, big.NewInt(0))
-		return err == nil
-	}, 500*time.Millisecond, 50*time.Millisecond)
+	returnedRates, err = ratesAdmin.Contract().GetRates(&bind.CallOpts{}, big.NewInt(0))
```

Comment:
`Contract()` returns a pointer so can be `nil`.

Reply:
That's true, but it's in a test so if it panics the test will fail and we'll know.